### PR TITLE
MODE-2549 Adds PDF sequencer

### PR DESCRIPTION
**Pull request by RobSis**
New pull request against the master branch. 

First review was done here:
https://github.com/ModeShape/modeshape/pull/1499

### CODE REVIEW

File: modeshape-parent/pom.xml [hunk 1]
```diff
@@ -181,6 +181,7 @@
         <version.org.jboss.ironjacamar>1.0.13.Final</version.org.jboss.ironjacamar>
         <version.com.datastax.cassandra>2.0.0-rc2</version.com.datastax.cassandra>
         <version.org.apache.cassandra>2.0.3</version.org.apache.cassandra>
+        <version.org.apache.pdfbox>1.8.12</version.org.apache.pdfbox>
```
Comment (hchiorean) [inline modeshape-parent/pom.xml]: can you please change this to be `1.8.10`.  ModeShape already uses Apache Tika for text and mime-type extraction which in turn uses this version:
```
 org.apache.tika:tika-parsers:jar:1.12:compile
 +- org.gagravarr:vorbis-java-tika:jar:0.6:compile
 +- org.apache.james:apache-mime4j-core:jar:0.7.2:compile
 +- org.apache.james:apache-mime4j-dom:jar:0.7.2:compile
 +- org.apache.commons:commons-compress:jar:1.10:compile
 +- org.tukaani:xz:jar:1.5:compile
 +- commons-codec:commons-codec:jar:1.9:compile
 +- org.apache.pdfbox:pdfbox:jar:1.8.10:compile
 |  +- org.apache.pdfbox:fontbox:jar:1.8.10:compile
 |  +- org.apache.pdfbox:jempbox:jar:1.8.10:compile
```

File: sequencers/modeshape-sequencer-pdf/src/main/java/org/modeshape/sequencer/pdf/PdfMetadataSequencer.java [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+                Calendar calendarValue = (Calendar) value;
+                if (calendarValue.getTimeZone().getID().toLowerCase().equals("unknown")) {
+                    calendarValue.setTimeZone(TimeZone.getDefault());
+                }
+                node.setProperty(propertyName, calendarValue);
+            } else if (value instanceof byte[]) {
+                InputStream is = new ByteArrayInputStream((byte []) value);
+                Binary binaryProperty = node.getSession().getValueFactory().createBinary(is);
+                node.setProperty(propertyName, binaryProperty);
+            } else if (value instanceof List) {
```
Comment (hchiorean) [inline sequencers/modeshape-sequencer-pdf/src/main/java/org/modeshape/sequencer/pdf/PdfMetadataSequencer.java]: if this uses unchecked types, should there be an `@SupressWarnings("unchecked")` in this method ?
Either that or change to `List<?>`

**Comment by hchiorean** [comment]
this looks good to me with some minor comments
@rhauch ?

**Comment by rhauch** [comment]
@hchiorean, this looks good to me as long as the above issues are addressed.

**Pull request merged**